packages/core/src/local-workspace/state.ts

```diff
@@ -105,30 +150,40 @@ export const localState = (filePath: string): state.State => {
       await inMemState.updatePathIndex(unmergedElements, servicesToMaintain)
       dirty = true
     },
-    rename: async (name: string): Promise<void> => {
-      const newFilePath = path.join(path.dirname(currentFilePath), `${name}${ZIPPED_STATE_EXTENSION}`)
-      await rename(currentFilePath, newFilePath)
-      currentFilePath = newFilePath
+    rename: async (newPrefix: string): Promise<void> => {
+      glob(`${currentFilePrefix}*${ZIPPED_STATE_EXTENSION}`, async (_err: Error | null, files: string[]) => {
+        files.forEach(async filename => {
+          const newFilePath = filename.replace(currentFilePrefix, path.join(path.dirname(currentFilePrefix), `${newPrefix}`))
+          await rename(filename, newFilePath)
+        })
+      })
+      currentFilePrefix = newPrefix
     },
     flush: async (): Promise<void> => {
       if (!dirty && pathToClean === '') {
         return
       }
-      const stateText = await createStateText()
-      await mkdirp(path.dirname(currentFilePath))
-      await replaceContents(currentFilePath, await generateZipString(stateText))
+      await mkdirp(path.dirname(currentFilePrefix))
+      const stateTextPerService = await createStateTextPerService()
+      Object.keys(stateTextPerService).forEach(async service => {
+        await replaceContents(`${currentFilePrefix}.${service}${ZIPPED_STATE_EXTENSION}`,
+          await generateZipString(stateTextPerService[service]))
+      })
       if (pathToClean !== '') {
         await rm(pathToClean)
       }
       log.debug('finish flushing state')
     },
     getHash: async (): Promise<string> => {
-      const stateText = await createStateText()
-      return toMD5(stateText)
+      const stateText = await createStateTextPerService()
+      return toMD5(safeJsonStringify(stateText))
     },
     clear: async (): Promise<void> => {
+      const existingServices = await inMemState.existingServices()
       await inMemState.clear()
-      await rm(currentFilePath)
+      existingServices.forEach(async service => {
+        await rm(`${currentFilePrefix}.${service}${ZIPPED_STATE_EXTENSION}`)
+      })
```

Comment:
IIUC this method will return a resolved promise before all of the files were deleted.
```suggestion
      await Promise.all(existingServices.map(service => 
          rm(`${currentFilePrefix}.${service}${ZIPPED_STATE_EXTENSION}`)
      }))
```